File: english-hindi-app/app/src/main/java/com/example/englishhindi/model/LessonWord.java
package com.example.englishhindi.model;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;

@Entity(tableName = "lesson_words",
        primaryKeys = {"lessonId", "wordId"},
        foreignKeys = {
            @ForeignKey(entity = Lesson.class,
                        parentColumns = "id",
                        childColumns = "lessonId",
                        onDelete = ForeignKey.CASCADE),
            @ForeignKey(entity = Word.class,
                        parentColumns = "id",
                        childColumns = "wordId",
                        onDelete = ForeignKey.CASCADE)
        },
        indices = {
            @Index("lessonId"),
            @Index("wordId")
        })
public class LessonWord {
    
    private int lessonId;
    private int wordId;
    private int order;
    private String notes;
    
    public LessonWord(int lessonId, int wordId, int order, String notes) {
        this.lessonId = lessonId;
        this.wordId = wordId;
        this.order = order;
        this.notes = notes;
    }
    
    // Getters and Setters
    public int getLessonId() {
        return lessonId;
    }
    
    public void setLessonId(int lessonId) {
        this.lessonId = lessonId;
    }
    
    public int getWordId() {
        return wordId;
    }
    
    public void setWordId(int wordId) {
        this.wordId = wordId;
    }
    
    public int getOrder() {
        return order;
    }
    
    public void setOrder(int order) {
        this.order = order;
    }
    
    public String getNotes() {
        return notes;
    }
    
    public void setNotes(String notes) {
        this.notes = notes;
    }
}

File: app/src/main/java/com/example/englishhindi/data/model/LessonWord.kt
package com.example.englishhindi.data.model

import androidx.room.Entity
import androidx.room.ForeignKey

/**
 * Junction entity creating many-to-many relationship between Lesson and Word.
 * Includes additional metadata about the word's position in the lesson.
 */
@Entity(
    tableName = "lesson_words",
    primaryKeys = ["lessonId", "wordId"],
    foreignKeys = [
        ForeignKey(
            entity = Lesson::class,
            parentColumns = ["id"],
            childColumns = ["lessonId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Word::class,
            parentColumns = ["id"],
            childColumns = ["wordId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class LessonWord(
    val lessonId: Long,
    val wordId: Long,
    
    // Word positioning and metadata
    val orderInLesson: Int, // Position of word in the lesson
    val isKeyword: Boolean = false, // Whether it's a focus word for the lesson
    val notes: String? = null, // Additional context for this word in this lesson
    
    // Content flags
    val includeInQuiz: Boolean = true, // Whether to include in lesson quiz
    val highlightInContent: Boolean = false // Whether to highlight in lesson content
)